src/measure_theory/integral/set_integral.lean

```diff
@@ -787,6 +787,105 @@ end
 
 end tendsto_mono
 
+/-! ### First moment method -/
+
+section first_moment
+variables {μ : measure α} {s N : set α} {f : α → ℝ} {r : ℝ}
+
+/-- **First moment method**. An integrable function is smaller than its mean on a set of positive
+measure. -/
+lemma measure_le_set_integral_pos (hμ : μ s ≠ 0) (hμ₁ : μ s ≠ ∞) (hf : integrable_on f s μ)
+  (hs : null_measurable_set s μ) :
+  0 < μ {x ∈ s | f x ≤ ∫ a in s, f a ∂μ / (μ s).to_real} :=
```

Comment:
Can you add the `lintegral` version as well?

Reply:
Combining this with @sgouezel's suggestion, should I define `laverage` for the Lebesgue average of an `ennreal`-valued function?

Reply:
I just want to say I don't have any opinion on defining `laverage` since I'm not sure of any cases where it is more convinent than writting the definition out explicitly.

Reply:
This turns out to be much more work than expected, so I am leaving a todo. You can have a look at the current state of things on branch `first_moment_lintegral`.